sdk/iot/Azure.Iot.Hub.Service/tests/DevicesClientTests.cs

```diff
@@ -2,66 +2,393 @@
+                Response<DeviceIdentity> createResponse = await client.Devices.CreateOrUpdateIdentityAsync(
+                    new Models.DeviceIdentity
+                    {
+                        DeviceId = testDeviceName
+                    }).ConfigureAwait(false);
+
+                device = createResponse.Value;
+
+                // Get device
+                // Get the device and compare ETag values (should remain unchanged);
+                Response<DeviceIdentity> getResponse = await client.Devices.GetIdentityAsync(testDeviceName).ConfigureAwait(false);
+
+                getResponse.Value.Etag.Should().BeEquivalentTo(device.Etag, "ETag value should not have changed.");
+
+                device = getResponse.Value;
+
+                // Update a device
+                device.Status = DeviceStatus.Disabled;
+
+                // TODO: (azabbasi) We should leave the IfMatchPrecondition to be the default value once we know more about the fix.
+                Response<DeviceIdentity> updateResponse = await client.Devices.CreateOrUpdateIdentityAsync(device, IfMatchPrecondition.UnconditionalIfMatch).ConfigureAwait(false);
+
+                updateResponse.Value.Status.Should().Be(DeviceStatus.Disabled, "Device should have been disabled");
+
+                // Delete the device
+                // Deleting the device happens in the finally block as cleanup.
+            }
+            finally
+            {
+                await Cleanup(client, device);
+            }
+        }
+
+        /// <summary>
+        /// Test basic operations of a Device Twin.
+        /// </summary>
+        [Test]
+        public async Task DevicesClient_DeviceTwinLifecycle()
+        {
+            string testDeviceName = $"TwinLifecycleDevice{GetRandom()}";
 
             DeviceIdentity device = null;
 
             IoTHubServiceClient client = GetClient();
+
             try
             {
-                try
-                {
-                    device = (await client.Devices.GetIdentityAsync(testDeviceName).ConfigureAwait(false)).Value;
-                }
-                catch (RequestFailedException ex) when (ex.Status == (int)HttpStatusCode.NotFound)
+                // Create a device
+                // Creating a device also creates a twin for the device.
+                Response<DeviceIdentity> createResponse = await client.Devices.CreateOrUpdateIdentityAsync(
+                    new Models.DeviceIdentity
+                    {
+                        DeviceId = testDeviceName
+                    }).ConfigureAwait(false);
+
+                device = createResponse.Value;
+
+                // Get the device twin
+                Response<TwinData> getResponse = await client.Devices.GetTwinAsync(testDeviceName).ConfigureAwait(false);
+                TwinData deviceTwin = getResponse.Value;
+
+                deviceTwin.DeviceId.Should().BeEquivalentTo(testDeviceName, "DeviceId on the Twin should match that of the device.");
```

Comment:
Q - why did we not call this `testDeviceId`, instead of `testDeviceName`?

Reply:
mainly because I think they are interchangeable.